juju/juju
# Add initial caas provisioner worker

## Description of change

Introduce the CAAS provioner worker and associated facades. Based on initial work by @mjs and @mikemccracken 

For now, any deployed charm will result in no-op operator running in a pod named after the application.

As is the case when adding new facades/workers, there's a lot of boilerplate. The key changes include:

- worker which (for now) deploys an operator running a stock Ubuntu image
- facade which allows to worker to watch  for new applications and ask for config
- restricted CAAS facade support for logins to CAAS models
- new manifold config for CAAS models
- CAAS model entity (includes moving CAAS model tests off ModelSuite)
- driveby fixes to stop errors on CAAS model destroy

## QA steps

juju deploy kubernetes-core
juju add-caas kubernetes caascloud
juju add-model mycaas caascloud
juju deploy <any charm>
kubectl get all

see that a new juju-operator-<appname> is created
